core/app/models/spree/exports/products.rb

```diff
@@ -24,7 +24,50 @@ def scope
       end
 
       def csv_headers
-        @csv_headers ||= Spree::CSV::ProductVariantPresenter::CSV_HEADERS + properties_headers
+        (
+          Spree::CSV::ProductVariantPresenter::CSV_HEADERS +
+          images_headers +
+          options_headers +
+          categories_headers +
+          properties_headers
+        ).flatten
+      end
+
+      def images_headers
```

Comment:
This all will set the number of headers based on data we're exporting in `records_to_export`